stanford_alt_check.forms.inc

```diff
@@ -2,124 +2,532 @@
+          $rows[] = array(
+            'Image' => image_style_url('thumbnail', $item['uri']),
+            'Title' => l(t($title), $host_url),
+            'Published' => $published,
+            'Alt Attribute' => $item['alt'],
+            'Last Changed' => time() - $host_entity->changed,
+            'type' => 'upload',
+            'delta' => $delta,
+            'original_src' => file_create_url($item['uri']),
+            'entity_info' => array(
+              'entity_type' => $entity_type,
+              'entity_id' => $entity_id,
+              'field_name' => $field['field_name'],
+              'host_type' => $host_type,
+              'host_id' => $host_id,
+              'host_url' => $host_url,
+            ),
+          );
+        }
+      }
+    }
+  }
+}
+
+/**
+ * Use the field and filter to do entity field queries and obtain a list.
+ *
+ * @param string $field
+ *   The field name on the entity.
+ * @param string|null $filter
+ *   If we want to filter out certain entities.
+ *
+ * @return array
+ *   The keyed array provided by EntityFieldQuery.
+ */
+function _stanford_alt_check_get_results($field, $filter = NULL) {
+  if ($filter == 'empty') {
+    // Get filled alt attributes to negate.
+    $filled_results = _stanford_alt_check_get_results($field, 'filled');
+
+    if (!$filled_results) {
+      // If no entities have filled in alt attributes, just give all entities
+      // with images.
+      return _stanford_alt_check_get_results($field);
+    }
+
+    $empty_results = array();
+    foreach (array_keys($filled_results) as $type) {
+
+      // Get all entities that do not have populated alt attributes.
+      $field_query = new EntityFieldQuery();
+      $field_query->entityCondition('entity_type', $type)
+        ->entityCondition('entity_id', array_keys($filled_results[$type]), 'NOT IN')
+        ->fieldCondition($field, 'fid', '0', '>=');
+
+      // Merge the results together.
+      foreach ($field_query->execute() as $type => $entities) {
+        if (!isset($empty_results[$type])) {
+          $empty_results[$type] = $entities;
+        }
+        else {
+          $empty_results[$type] = array_merge($empty_results[$type], $entities);
+        }
+      }
+    }
+
+    return $empty_results;
+  }
+  elseif ($filter == 'filled') {
+    // Only entites with populated image alt attributes.
+    $field_query = new EntityFieldQuery();
+    $field_query->fieldCondition($field, 'fid', '0', '>=');
+    $field_query->fieldCondition($field, 'alt', '', '!=');
+    return $field_query->execute();
+  }
+
+  // All entities with images.
+  $field_query = new EntityFieldQuery();
+  $field_query->fieldCondition($field, 'fid', '0', '>=');
+  return $field_query->execute();
+}
+
+/**
+ * Get all fields of image type.
+ *
+ * @return array
+ *   All image fields.
+ */
+function _stanford_alt_check_upload_fields() {
+  $fields = array();
+  foreach (field_info_fields() as $field) {
```

Comment:
For a micro performance improvement you can probably use `field_info_field_map()` instead of `field_info_fields()`